Molio/Source/Presentation/SelectPlaylist/SelectPlaylistViewModel.swift

```diff
@@ -0,0 +1,59 @@
+import Combine
+import Foundation
+
+final class SelectPlaylistViewModel: ObservableObject {
+    @Published var playlists: [MolioPlaylist] = []
+    @Published var selectedPlaylist: MolioPlaylist?
+    
+    private let fetchAllPlaylistsUseCase: FetchAllPlaylistsUseCase
+    private let changeCurrentPlaylistUseCase: ChangeCurrentPlaylistUseCase
+    private let publishCurrentPlaylistUseCase: PublishCurrentPlaylistUseCase
+    private var cancellables = Set<AnyCancellable>()
+    
+    init(
+        fetchAllPlaylistsUseCase: FetchAllPlaylistsUseCase = DefaultFetchAllPlaylistsUseCase(),
+        changeCurrentPlaylistUseCase: ChangeCurrentPlaylistUseCase = DefaultChangeCurrentPlaylistUseCase(),
+        publishCurrentPlaylistUseCase: PublishCurrentPlaylistUseCase = DIContainer.shared.resolve()
+    ) {
+        self.fetchAllPlaylistsUseCase = fetchAllPlaylistsUseCase
+        self.changeCurrentPlaylistUseCase = changeCurrentPlaylistUseCase
+        self.publishCurrentPlaylistUseCase = publishCurrentPlaylistUseCase
+        
+        reloadData()
+    }
+
+    func reloadData() {
+        bindPublishCurrentPlaylist()
+        fetchPlaylists()
+    }
+
+    func selectPlaylist(_ playlist: MolioPlaylist) {
+        selectedPlaylist = playlist
+    }
+    
+    func savePlaylist () {
+        guard let selectedPlaylist else { return }
+        changeCurrentPlaylistUseCase.execute(playlistId: selectedPlaylist.id)
+    }
+
+    // MARK: - Private Method
+    
+    private func fetchPlaylists() {
+        Task {
+            let playlists = await fetchAllPlaylistsUseCase.execute()
+            DispatchQueue.main.async {
+                self.playlists = playlists
+            }
+        }
```

Comment:
```suggestion
        Task { @MainActor in
            self.playlists = await fetchAllPlaylistsUseCase.execute()
        }
```
이렇게 사용하시면 어떨까요? 어차피 await 뒤에 있는 fetchAllPlaylistsUseCase.execute는 다른 스레드 (Main이 아닌 스레드)에서 실행됩니다.

Reply:
viewModel 과 이 코드가 연결되어 있어서인지 메인 스레드에서 실행된다는 오류가 발생하여 이렇게 작성하였습니다. 다시 알아보겠습니다.

Reply:
창우님 코드 적용했습니다! mainActor에 대해 공부해보겠습니다

Reply:
감사합니다!!